Mage/src/main/java/mage/abilities/effects/common/CopyEffect.java

```diff
@@ -80,13 +80,10 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
         Permanent permanent = affectedObjectList.get(0).getPermanent(game);
-        if (permanent == null) {
-            permanent = (Permanent) game.getLastKnownInformation(getSourceId(), Zone.BATTLEFIELD, source.getSourceObjectZoneChangeCounter());
-            // As long as the permanent is still in the LKI continue to copy to get triggered abilities to TriggeredAbilities for dies events.
-            if (permanent == null) {
-                discard();
-                return false;
-            }
+        if (permanent == null && !game.getShortLivingLKI(getSourceId(), Zone.BATTLEFIELD)) {
+            // As long as the permanent is still in the short living LKI continue to copy to get triggered abilities to TriggeredAbilities for dies events.
+            discard();
+            return false;
```

Comment:
If present in short living LKI, then don't you still need to copy from the LKI?

Reply:
Oh right, I'll try to make a test that would have caught that for the day we eventually remove short lki.

Reply:
Ok I couldn't have a test catching that, but now the permanent is set back to the LKI one if getShortLivingLKI returns true.